student/src/main/java/tds/student/services/remote/ExamStatusMapper.java

```diff
@@ -0,0 +1,40 @@
+package tds.student.services.remote;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import tds.exam.ExamStatusCode;
+import tds.student.sql.data.OpportunityStatusType;
+
+class ExamStatusMapper {
+  private static final Map<String, OpportunityStatusType> examStatusToOpportunityStatus = new HashMap<>();
+
+  static {
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_APPROVED, OpportunityStatusType.Approved);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_FAILED, OpportunityStatusType.Failed);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_PAUSED, OpportunityStatusType.Paused);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_PENDING, OpportunityStatusType.Pending);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_REVIEW, OpportunityStatusType.Review);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_STARTED, OpportunityStatusType.Started);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_SUSPENDED, OpportunityStatusType.Suspended);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_DENIED, OpportunityStatusType.Denied);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_COMPLETED, OpportunityStatusType.Completed);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_SCORED, OpportunityStatusType.Scored);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_SEGMENT_ENTRY, OpportunityStatusType.SegmentEntry);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_SEGMENT_EXIT, OpportunityStatusType.SegmentExit);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_CLOSED, OpportunityStatusType.Closed);
+    examStatusToOpportunityStatus.put(ExamStatusCode.STATUS_DISABLED, OpportunityStatusType.Disabled);
+  }
+
+  static OpportunityStatusType parseExamStatus(final String examStatusCode) {
+    OpportunityStatusType parsedStatusType = OpportunityStatusType.Unknown;
+
+    if (StringUtils.isNotEmpty(examStatusCode)) {
+      parsedStatusType = examStatusToOpportunityStatus.get (examStatusCode);
```

Comment:
I know you copied this over, but seems like the check should be on `examStatusToOpportunityStatus.containsKey(examStatusCode)` otherwise if it doesn't exist is `parsedStatusType` null or does the call to `get` throw?